monero-sys/src/lib.rs

```diff
@@ -123,16 +163,60 @@ pub struct Daemon {
     pub ssl: bool,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[typeshare]
+pub struct TransactionInfo {
+    #[serde(with = "monero_serde")]
+    pub fee: monero::Amount,
+    #[serde(with = "monero_serde")]
+    pub amount: monero::Amount,
+    #[typeshare(serialized_as = "number")]
+    pub confirmations: u64,
+    pub tx_hash: String,
+    pub direction: TransactionDirection,
+}
+
+#[typeshare]
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub enum TransactionDirection {
+    In,
+    Out,
+}
+
 /// A wrapper around a pending transaction.
 pub struct PendingTransaction(*mut ffi::PendingTransaction);
 
+/// A struct containing transaction history.
+pub struct TransactionHistoryHandle(*mut TransactionHistory);
+
+/// A struct containing a single transaction.
+pub struct TransactionInfoHandle(*mut ffi::TransactionInfo);
+
+// Safety: The underlying C++ object is thread-safe for immutable access once obtained from TransactionHistory.
```

Comment:
Which underlying object? TransactionHistoryHandle contains a mutable TransactionHistory pointer. I seriously doubt that the `Sync` impl is safe

Reply:
That comment is AI

Reply:
Figured xD We have to review these impl's in detail before merging, possibly we can't have either.